_extensions/redirects.py

```diff
@@ -0,0 +1,55 @@
+# Adapted from https://github.com/sphinx-contrib/redirects
+
+import os
+import re
+from pathlib import Path
+
+from sphinx.builders import html as builders
+from sphinx.util import logging as logging
+
+TEMPLATE = '<html><head><meta http-equiv="refresh" content="0; url=%s"/></head></html>'
+
+logger = logging.getLogger(__name__)
+
+def generate_redirects(app):
+    path = os.path.join(app.srcdir, app.config.redirects_file)
+    if not os.path.exists(path):
+        logger.info("Could not find redirects file at '%s'", path)
+        return
+
+    source_suffix = next(iter(app.config.source_suffix))
+
+    if not type(app.builder) == builders.StandaloneHTMLBuilder:
+        logger.info("Redirects are only supported by the 'html' builder. Skipping...")
+        return
+
+    with open(path) as redirects:
+        pattern = re.compile(r'^([\w\-./ ]+)(#.*)?$')
+        for line in redirects.readlines():
+            # Exclude comment or empty lines
+            match_result = pattern.search(line)
+            if not match_result:
+                continue
+
+            # Parse the rule
+            redirect = match_result.group(1).rstrip()
+            if redirect.count(' ') != 1:
```

Comment:
this can be avoided with above regex

Reply:
I extracted the `[ ]*` as suggested above but I think that it's better to warn the user if he wrote something that looks like a rule but it malformed.

For instance, I wouldn't be surprise that someone writes `sales/my_new_page.rst  # This is the new file` instead of `sales/old_page.rst sales/my_new_page.rst`.

What do you think ?

Reply:
Indeed, if we don't match the regex, we should log an error "please use format `source.rst target.rst`. Just that we don't need to make the job that can done with a regex.
By the way, we can direct include the `.rst` extension in the regex too :+1:

Reply:
Fine for me :+1: